api/api.go

```diff
@@ -75,226 +77,216 @@ func (api *ImportAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	jobID := vars["id"]
 	logData := log.Data{jobIDKey: jobID}
 	auditParams := common.Params{jobIDKey: jobID}
+	ctx := r.Context()
 
-	if err := api.auditor.Record(r.Context(), uploadFileAction, actionAttempted, auditParams); err != nil {
-		handleAuditError(w, err, logData)
+	if auditError := api.auditor.Record(ctx, uploadFileAction, actionAttempted, auditParams); auditError != nil {
+		handleAuditError(ctx, w, uploadFileAction, actionAttempted, auditError, logData)
 		return
 	}
 
-	uploadedFile, err := models.CreateUploadedFile(r.Body)
-	defer r.Body.Close()
-	if err != nil {
-		log.Error(err, logData)
-		http.Error(w, "bad client request received", http.StatusBadRequest)
-		return
-	}
-
-	logData["file"] = uploadedFile
-	auditParams["fileAlias"] = uploadedFile.AliasName
-	auditParams["fileURL"] = uploadedFile.URL
-
-	if err := api.dataStore.AddUploadedFile(jobID, uploadedFile); err != nil {
-		if err := api.auditor.Record(r.Context(), uploadFileAction, actionUnsuccessful, auditParams); err != nil {
-			handleAuditError(w, err, logData)
+	options := make(map[string]bool)
+	if err := api.addUploadFile(ctx, r, jobID, options, auditParams, logData); err != nil {
+		if auditError := api.auditor.Record(ctx, uploadFileAction, actionUnsuccessful, auditParams); auditError != nil {
+			handleAuditError(ctx, w, uploadFileAction, actionUnsuccessful, auditError, logData)
 			return
 		}
-		log.ErrorC("error while adding up", err, logData)
-		setErrorCode(w, err)
+
+		setErrorCode(w, err, options)
 		return
 	}
-	if err := api.auditor.Record(r.Context(), uploadFileAction, actionSuccessful, auditParams); err != nil {
-		handleAuditError(w, err, logData)
+
+	if auditError := api.auditor.Record(ctx, uploadFileAction, actionSuccessful, auditParams); auditError != nil {
+		auditActionFailure(ctx, uploadFileAction, actionSuccessful, auditError, logData)
 		return
 	}
-	log.Info("added uploaded file to job", log.Data{jobIDKey: jobID, "file": uploadedFile})
+
+	audit.LogInfo(ctx, "added uploaded file to job", logData)
 }
 
-func (api *ImportAPI) updateJob(w http.ResponseWriter, r *http.Request) {
+func (api *ImportAPI) updateJobHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
 	vars := mux.Vars(r)
 	jobID := vars["id"]
 	logData := log.Data{jobIDKey: jobID}
 	auditParams := common.Params{jobIDKey: jobID}
 
-	if err := api.auditor.Record(r.Context(), updateJobAction, actionAttempted, auditParams); err != nil {
-		handleAuditError(w, err, logData)
+	if auditError := api.auditor.Record(ctx, updateJobAction, actionAttempted, auditParams); auditError != nil {
+		handleAuditError(ctx, w, updateJobAction, actionAttempted, auditError, logData)
 		return
 	}
 
-	job, err := models.CreateJob(r.Body)
-	defer r.Body.Close()
-	if err != nil {
-		log.Error(err, logData)
-		http.Error(w, "bad client request received", http.StatusBadRequest)
-		return
-	}
+	options := make(map[string]bool)
+	if err := func(options map[string]bool) (err error) {
```

Comment:
Could you refactor this so it isn't inline too?

Reply:
nope ;)